query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -493,44 +492,44 @@ class StdVarOverTimeFunction(var sum: Double = 0d,
   }
 }
 
-abstract class VarOverTimeChunkedFunctionD(var sum: Double = 0d,
+abstract class VarOverTimeChunkedFunctionD(var sum: Double = Double.NaN,
                                            var count: Int = 0,
-                                           var squaredSum: Double = 0d) extends ChunkedDoubleRangeFunction {
-  override final def reset(): Unit = { sum = 0d; count = 0; squaredSum = 0d }
+                                           var squaredSum: Double = Double.NaN,
+                                           var lastSample: Double = Double.NaN) extends ChunkedDoubleRangeFunction {
+  override final def reset(): Unit = { sum = Double.NaN; count = 0; squaredSum = Double.NaN }
   final def addTimeDoubleChunks(doubleVectAcc: MemoryReader,
                                 doubleVect: BinaryVector.BinaryVectorPtr,
                                 doubleReader: bv.DoubleVectorDataReader,
                                 startRowNum: Int,
                                 endRowNum: Int): Unit = {
     val it = doubleReader.iterate(doubleVectAcc, doubleVect, startRowNum)
-    var _sum = 0d
-    var _sqSum = 0d
     var elemNo = startRowNum
     while (elemNo <= endRowNum) {
       val nextValue = it.next
       if (!JLDouble.isNaN(nextValue)) {
-        _sum += nextValue
-        _sqSum += nextValue * nextValue
-        elemNo += 1
+        if (sum.isNaN()) sum = 0d
+        if (squaredSum.isNaN()) squaredSum = 0d
+        if (elemNo == endRowNum) lastSample = nextValue
+        sum += nextValue
+        squaredSum += nextValue * nextValue
+        count +=1
       }
+      elemNo += 1
     }
-    count += (endRowNum - startRowNum + 1)
-    sum += _sum
-    squaredSum += _sqSum
   }
 }
 
 class StdDevOverTimeChunkedFunctionD extends VarOverTimeChunkedFunctionD() {
   final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
-    val avg = if (count > 0) sum/count else 0d
+    val avg = sum/count
```

Comment:
Here too, I think retaining divide by 0 is good idea

Reply:
Same reason as above.

Reply:
Added the check @velvia . Another issue I found is that in stddev, stdvar and zscore, that the count check is done just for computing avg and not for computing stddev & stdvar which is in the newline above the line you pointed here. The fix I added takes care of all now. Thanks for pointing out.